test/parallel/test-stream-construct.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+
+const common = require('../common');
+const { Writable, Readable } = require('stream');
+const assert = require('assert');
+
+{
+  // Pass options as is to construct.
+
+  const options2 = {
+    test: 'foo',
+    construct: common.mustCall((options, callback) => {
+      assert.strictEqual(options, options2);
+      callback();
+    })
+  };
+  const w = new Writable(options2);
+  w.destroy();
+}
+
+{
+  // Pass options as is to construct.
+
+  const options2 = {
+    test: 'foo',
+    construct: common.mustCall((options, callback) => {
+      assert.strictEqual(options, options2);
+      callback();
+    }),
+    read() {}
+  };
+  const r = new Readable(options2);
+  r.destroy();
+}
+
+{
+  // Multiple callback.
+  new Writable({
+    construct: common.mustCall((options, callback) => {
+      callback();
+      callback();
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    code: 'ERR_MULTIPLE_CALLBACK'
+  }));
+}
+
+{
+  // Multiple callback.
+  new Readable({
+    construct: common.mustCall((options, callback) => {
+      callback();
+      callback();
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    code: 'ERR_MULTIPLE_CALLBACK'
+  }));
+}
+
+{
+  // Synchronous error.
+
+  new Writable({
+    construct: common.mustCall((options, callback) => {
+      callback(new Error('test'));
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    message: 'test'
+  }));
+}
+
+{
+  // Synchronous error.
+
+  new Readable({
+    construct: common.mustCall((options, callback) => {
+      callback(new Error('test'));
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    message: 'test'
+  }));
+}
+
+{
+  // Asynchronous error.
+
+  new Writable({
+    construct: common.mustCall((options, callback) => {
+      process.nextTick(callback, new Error('test'));
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    message: 'test'
+  }));
+}
+
+{
+  // Asynchronous error.
+
+  new Readable({
+    construct: common.mustCall((options, callback) => {
+      process.nextTick(callback, new Error('test'));
+    })
+  }).on('error', common.expectsError({
+    type: Error,
+    message: 'test'
+  }));
+}
+
+{
+  // Throw synchronously.
+
+  common.expectsError(function() {
+    new Writable({
+      construct: common.mustCall((options, callback) => {
+        throw new Error('test');
+      })
+    });
+  }, {
+    type: Error,
+    message: 'test'
+  });
+}
+
+{
+  // Throw synchronously.
+
+  common.expectsError(function() {
+    new Readable({
+      construct(options, callback) {
+        throw new Error('test');
+      }
+    });
+  }, {
+    type: Error,
```

Comment:
:point_up: the `type` property was removed and `common.expectsError` has to be changed to `assert.throws` in this case.